Fixed #8760 -- Changed ModelMultipleChoiceField to use invalid_list as a error message key.

smithdc1 opened the pull request:
[Ticket](https://code.djangoproject.com/ticket/8760)

This longstanding issue had a patch provided by some years ago. However, Tim raised the question about if and how this should be implemented. e.g. if we accept this change is it ok as a backwards incompatible change or do we try a deprecation path. 

I've put together this PR to help this discussion. Making this change only broke one minor test which I've provided a fix for, hopefully this can help the debate. 

I'm not sure if this is a simple decision that can be made here or if wider discussion (on the mailing list) is warranted. 

Appreciate your views & guidance.

felixxm commented:
This error message key is documented and tested, so we should deprecate it properly and removed it in Django 4.0. We can raise an error when it's provided, e.g.
```python
class ModelMultipleChoiceField(ModelChoiceField):
    ...

    def __init__(self, queryset, **kwargs):
        super().__init__(queryset, empty_label=None, **kwargs)
        if self.error_messages.get('list') is not None:
            warnings.warn(...)
            self.error_messages['invalid_list'] = self.error_messages['list']
```

smithdc1 commented:
Hi @felixxm 

Thank you very much for your guidance. 

I've updated the `ModelMultipleChoiceField` to raise a warning as you suggest. I've also made changes to docs, release notes & deprecation list. Finally I've added a test case to show the warning being raised when an instance of the class is created with the old 'list' error message. 

I therefore think this is now ready for another review. 

Thanks

felixxm commented:
@smithdc1 Thanks :+1: I pushed minor edits to docs and moved deprecation tests to a separate class (it's easier to remove them when deprecation ends).
